nuxeo-features/nuxeo-admin-center/nuxeo-connect-update/src/test/java/org/nuxeo/connect/download/tests/SlowDownloaderServlet.java

```diff
@@ -41,15 +41,15 @@ public void init() throws ServletException {
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         resp.setContentType("text/plain");
-
-        PrintWriter writer = resp.getWriter();
-
-        for (int i = 1; i < 10; i++) {
-            writer.write("Yoooooo\n");
-            try {
-                Thread.sleep(100);
-            } catch (InterruptedException e) {
-                // NOP
+        try (PrintWriter writer = resp.getWriter()) {
+            for (int i = 1; i < 10; i++) {
+                writer.write("Yoooooo\n");
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException(e);
```

Comment:
??? This is a test for a slow download, I am wondering if we need to throw the RuntimeException or just continue the loop?

Reply:
need to throw, there's no reason to swallow the interrupt and continue looping

Reply:
Yes `InterruptedException` only happens if someone wants to kill a thread, so we should never prevent that by ignoring it.